### DISCUSSION TITLE
problem about logo printing

### DISCUSSION THREAD

**Discussion by sniper-jawadino** [lukevp/ESC-POS-.NET#232] [General]
can you help me

please watch photo
 https://i.stack.imgur.com/ZE8NS.jpg 

everything's good just i have a problem about logo printing

is the code correct?

`var ff = new EPSON();

        printer.WriteAsync( // or, if using and immediate printer, use await printer.WriteAsync
            ByteSplicer.Combine(
            ff.CenterAlign(),
            ff.PrintImage(File.ReadAllBytes("Images/logo.png"), true),

            ff.PrintLine(""),

            ff.SetStyles(PrintStyle.FontB),
            ff.PrintLine("1   AAAAAAAAAAAAAAAAAAAAAAAAAAAA     89.95         89.95"),
            ff.PrintLine("--------------------------------------------------------"),

            ff.RightAlign(),
            ff.PrintLine("SUBTOTAL         89.95"),
            ff.PrintLine("Total Order:         89.95"),
            ff.PrintLine("Total Payment:         89.95"),
            ff.PrintLine(""),

            ff.PrintLine(""),
            ff.PrintLine(""),
 
          

          )
`

**Comment by lukevp** [comment]
Most of the time when images do not print properly, it's because the printer you're using doesn't fully support the latest ESC/POS image standard. If you're using a non-epson printer, try using the legacy image print instead. If you haven't yet run the console test app included in this project, I would recommend running it. It will let you test your printer with known-good code to make sure your printer and drivers are set up properly. There are both image printing modes included in the program.

Another note - if you are using WriteAsync, you need to also await it. In C# it's standard to end methods with Async to indicate they must be awaited.

**Comment by sniper-jawadino** [comment]
Thank you very much for your feedback

after test with (ESCPOS_NET.ConsoleTest)
I found that the (LegacyImages) mode works very well  (I added TRUE on the line)


I have the last concern
how to access photos on android directory
 
![dcdcdc](https://github.com/lukevp/ESC-POS-.NET/assets/128584772/12342d33-14f7-4d35-9816-fc88788d549d)




Thanks for your help

**Reply by lukevp** [reply]
Are you using Xamarin or MAUI? This isn't really a problem specific to this library. Once you read the byte array in, you can send it into the library, so you should look more generally on a guide for how to store and read images in your platform.

The typical way this works with mobile development is that you can either bundle the images up into resources, or you can store them in an application-specific directory. Basically you'll need to solve how to 1) publish the file in your application bundle and 2) get a reference to its path and read it in, then 3) you send the bytes to the ESCPOS_NET library.

**Reply by sniper-jawadino** [reply]
I used MAUI

yes I know that the problem is very far with the ESCPOS_NET library

I will follow your advice
and we will look for how to transform an image to Bytes (MAUI)

thanks a lot for your help
